### PR TITLE
Autogenerate katacoda buildpack author guide

### PR DESCRIPTION
**Pull request by AidanDelaney**
Generate the katacoda scenarios for the buildpack author guide.  Also
bump to latest `pack` and api version.

### CODE REVIEW

**Comment by samj1912** [comment]
The generated katacoda docs should be similar to https://github.com/buildpacks/katacoda-scenarios/tree/main/buildpack-author-guide

I see we are missing a few editor integrations missing - namely `{{open}}` katacoda directives and the `<pre data-filename=... data-target=../>` directives.

I also see that the output `{{execute}}` directives seem a bit misplaced in a few places in the generated docs.

If there are some common patterns/substitutions, we can also update our template generator go script to do the right things.

**Comment by mboldt** [comment]
Just to make the link, this will resolve #438, right?

**Comment by AidanDelaney** [comment]
@mboldt Yes.  I'll address comments and complete #438 next week.

**Comment by AidanDelaney** [comment]
`pack inspect-image test-ruby-app --bom` no longer shows an actual BOM when using API 0.7.  Not sure what to replace it with.

**Comment by AidanDelaney** [comment]
https://www.katacoda.com/aidandelaney/scenarios/buildpack-author-guide should be a runnable version of these katacoda generated docs.

**Comment by samj1912** [comment]
@AidanDelaney thanks for this PR! The generation tools looks great 👍 Could we separate out the buildpack API bump from the katacoda generation. It will make it easier to review the changes related to the generation tooling from the changes introduced because of the API bump. The API bump should probably go in after @aemengo's PRs about download SBOM on the pack side and the docs side are merged. It is also easier t

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 1]
```diff
@@ -32,27 +32,60 @@ Buildpacks:
 
 Processes:
   TYPE                 SHELL        COMMAND                        ARGS
-  web (default)        bash         bundle exec ruby app.rb        
+  web (default)        bash         bundle exec ruby app.rb
   worker               bash         bundle exec ruby worker.rb
-```
+``` -->
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:37]: Should this be commented out? Or is this a leftover from testing?

File: content/docs/buildpack-author-guide/create-buildpack/make-app-runnable.md [hunk 2]
```diff
@@ -21,7 +22,7 @@ EOL
 # ...
 ```
 
-Your full `ruby-buildpack/bin/build` script should now look like the following:
+Your full `ruby-buildpack/bin/build`<!--+ "{{open}}" +--> script should now look like the following:
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/make-app-runnable.md:25]: Were you able to test the hugo output with these tags being placed right next to the code block? IIRC I had some difficulties placing a comment tag right after a code block.

**Review by dgsdoug — APPROVED** [review]

File: content/docs/buildpack-author-guide/create-buildpack/make-app-runnable.md (hunk 2 above)
Comment (AidanDelaney) [reply content/docs/buildpack-author-guide/create-buildpack/make-app-runnable.md:25]: The `<!--+ "{{open}}" +-->` comments work with hugo, but I had to modify the placement of all `<!--+ "{{execute}}" +-->` comments.

File: katacoda/scenarios/app-journey/index.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     }
   },
   "environment": {
-    "uilayout": "terminal"
+    "uilayout": "vscode-terminal-split"
```
Comment (samj1912) [inline katacoda/scenarios/app-journey/index.json:23]: Accidental?

Comment (AidanDelaney) [reply katacoda/scenarios/app-journey/index.json]: Yes, and reverted.  Thanks.

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 4]
```diff
@@ -14,14 +14,14 @@ One of the benefits of buildpacks is they can also populate the app image with m
...
-
 <!-- test:exec -->
 ```bash
 pack inspect-image test-ruby-app
 ```
+<!--+ "{{execute}}" +-->
 You should see the following:
 
-<!-- test:assert=contains -->
+<!-- test:assert=contains;ignore-lines=... -->
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:24]: do we need the ignore lines directive?

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 5]
```diff
@@ -14,14 +14,14 @@ One of the benefits of buildpacks is they can also populate the app image with m
 
 You can find all of this information using `pack` via its `inspect-image` command.
 
-
 <!-- test:exec -->
 ```bash
 pack inspect-image test-ruby-app
 ```
+<!--+ "{{execute}}" +-->
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:21]:
```suggestion
<!--+- "{{execute}}" +-->
```

this and others would need to strip whitespaces

Comment (AidanDelaney) [reply content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md]: I've normalized all the `<!--+ "{{execute}}" +-->` as `<!--+- "{{execute}}"+-->` and `<!--+ "{{open}}" +-->` as `<!--+"{{open}}"+-->`.

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md (hunk 4 above)
Comment (AidanDelaney) [reply content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md]: Removed

File: katacoda/scenarios/app-developer-guide/background.sh [hunk 6]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
-(curl -sSL "https://github.com/buildpacks/pack/releases/download/v0.18.1/pack-v0.18.1-linux.tgz" | sudo tar -C /usr/local/bin/ --no-same-owner -xzv pack)
+(curl -sSL "https://github.com/buildpacks/pack/releases/download/v0.23.0/pack-v0.23.0-linux.tgz" | sudo tar -C /usr/local/bin/ --no-same-owner -xzv pack)
 git clone https://github.com/buildpacks/samples
 sudo touch /opt/setup-done
 docker pull cnbs/sample-builder:bionic
 docker pull buildpacksio/lifecycle
+pack config trusted-builders add cnbs/sample-builder:bionic
```
Comment (samj1912) [inline katacoda/scenarios/app-developer-guide/background.sh:7]: Why was this changed?

File: katacoda/scenarios/app-developer-guide/using-cache-image.md [hunk 7]
```diff
@@ -75,4 +75,5 @@ pack build localhost:5000/buildpack-examples/second-cache-image-example \
     --publish
 ```{{execute}}
 
-[samples]: https://github.com/buildpack/samples
\ No newline at end of file
+[samples]: https://github.com/buildpack/samples
+---
```
Comment (samj1912) [inline katacoda/scenarios/app-developer-guide/using-cache-image.md:79]: Why is it adding these additional `---` at the end?

**Review by samj1912 — APPROVED** [review]
Lgtm. Suggested some minor fixes.

File: katacoda/scenarios/app-developer-guide/using-cache-image.md (hunk 7 above)
Comment (AidanDelaney) [reply katacoda/scenarios/app-developer-guide/using-cache-image.md]: I've cleaned this up now.  It was working around a rendering issue on katacoda.  Issue no-longer appears.
